inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1412,30 +1413,76 @@ case name:\
 }
 
 void GNAGraphCompiler::PermutePrimitive(InferenceEngine::CNNLayerPtr layer) {
+    if (LayerInfo(layer).isTrivialPermute()) {
+        return;
+    }
     auto layerOrder = layer->GetParamAsInts("order");
+    auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
+    auto inputs = layer->insData.begin()->lock();
+    auto inputsOrder = inputs->getTensorDesc().getDims();
+    auto outputs = layer->outData.front();
+
+    // squeeze order vector
+    SizeVector squeezedInputOrder;
+    for (auto input_shape : inputsOrder) {
+        if (input_shape != 1) squeezedInputOrder.push_back(input_shape);
+    }
+    SizeVector squeezedOutputOrder;
+    for (auto output_shape : layerOrder) {
+        if (output_shape != 0) squeezedOutputOrder.push_back(output_shape);
+    }
+
+    void* ptr_inputs = nullptr;
+    void* ptr_outputs = nullptr;
+
+    if (squeezedInputOrder.size() > 2) {
+        THROW_GNA_EXCEPTION << "unsupported permute (requested transpose is not 2D)";
+    }
 
-    string dimMessage;
-    if (layerOrder == vector<int>({0, 3, 2, 1})) {
-        return;  // supported case
+    if (std::min(squeezedInputOrder[0], squeezedInputOrder[1]) > 8) {
+        THROW_GNA_EXCEPTION << "unsupported permute (minor dimension="
```

Comment:
Please change it to `THROW_GNA_LAYER_EXCEPTION` so layer name will be printed to log